agent/local/state.go

```diff
@@ -366,7 +366,7 @@ func (l *State) SetServiceState(s *ServiceState) {
 }
 
 func (l *State) setServiceStateLocked(s *ServiceState) {
-	s.WatchCh = make(chan struct{})
+	s.WatchCh = make(chan struct{}, 1)
```

Comment:
Hmm I wonder if we rely on sync behaviour anywhere currently?

That said, see my comment above, do you need this if you use `Notify` mechanism?

Reply:
WatchCh was being used exclusively for notifications to the blocking query done by the `AgentService` handler here:
https://github.com/hashicorp/consul/blob/c10f83ebee52e4bcec50a99539a2e540ac7eafed/agent/agent_endpoint.go#L278

Previously those notifications relied on the channel being [closed](https://github.com/hashicorp/consul/blob/c10f83ebee52e4bcec50a99539a2e540ac7eafed/agent/local/state.go#L304) to notify about service updates/removals. 

Now I'm also sending messages in whenever a check is added/removed. So to avoid synchronizing I made this a buffered channel, and the channel sends are non-blocking. 

A reason why I'm reusing this channel is that now both `AgentService` and the service-checks cachetype use the same `LocalBlockingQuery`: https://github.com/hashicorp/consul/blob/5a53d6795a13e2b8d679c3b343fed6e692e1a825/agent/agent.go#L3472